### ISSUE TITLE
Shift-selecting  and removing default keyfiles in the mount parameters removes incorrect keyfiles, crashes wizard

### ISSUE THREAD

**Issue by seanbethard** [veracrypt/VeraCrypt#1342]
Hello!

  I think I found a bug in iCloud Soup that can be reproduced with the VeraCrypt Volume Creation Wizard by setting some default keyfiles (hundreds of them) in the mount parameters, shift-selecting some subset of them and removing them. I found that it was not possible to shift-select all ranges of keyfiles (such as those that begin with the first keyfile) and that the subset that was removed with the Remove button was not the selected subset. The error then follows.

It's unclear whether you must be signed in with your iTunes ID in order to reproduce the error.

Sean

A critical error has occurred and VeraCrypt must be terminated. If this is caused by a bug in VeraCrypt, we would like to fix it. To help us, you can send us an automatically generated error report containing the following items. Do you want to send us the error report?
- Program version
- Operating system version
- Hardware architecture
- Checksum of VeraCrypt executable
```
% md5 VeraCrypt                          
MD5 (VeraCrypt) = 3b2ca3aaf1bd000355802cb467274e5e
% shasum -a 256 VeraCrypt 
5e6087f10afd812fe82b20eac642dff322bf648a27b0f60ae4e0cd8ffeae2f48  VeraCrypt
```
- Error category (?)
- Error address (?)
- VeraCrypt call stack (?)
If you select 'Yes', the following URL (which contains the entire error report) will be opened in your default Internet browser.
https://www.veracrypt.fr

The URL that loads (without an error report):

https://veracrypt.fr/en/Home.html

System info:
  macOS: 14.4.1 (23E224)
  Model Name:    MacBook Pro
  Model Identifier:    MacBookPro17,1
  Model Number:    MYD92LL/A
  Chip:    Apple M1
  Total Number of Cores:    8 (4 performance and 4 efficiency)
  Memory:    8 GB
  System Firmware Version:    10151.101.3
  OS Loader Version:    10151.101.3

VeraCrypt 1.26.7

**Comment by Jertzukka** [comment]
Good catch. After looking into it, it seems to be an upstream issue in wxWidgets 3.2.2.1 which I believe the official binaries are built against. This was fixed in https://github.com/wxWidgets/wxWidgets/commit/d6dd0ebb6e25ac4fbde7f3cca2074be42a54c10d and after building against wxWidgets 3.2.4 I couldn't get it to crash anymore and the issues with the ListCtrl indices were gone.

To fix this for yourself, you'd have to build VeraCrypt yourself, or wait until new binaries or the next version is released which is hopefully built against a wxWidgets version which has this issue fixed.
